.github/scripts/auto-label.cjs

```diff
@@ -0,0 +1,165 @@
+  'eslint/': 'style',
+  '.eslintrc.js': 'style',
+  '.github/workflows/': 'ci',
+  '.github/scripts/': 'ci',
+
+};
+
+const getLabels = async (github, owner, repo, pullNumber) => {
+  const httpClient = new HttpClient();
+  const { data: pullRequest } = await github.rest.pulls.get({
+    owner,
+    repo,
+    // eslint-disable-next-line camelcase
+    pull_number: pullNumber,
+  });
+  if (!pullRequest.mergeable) {
+    console.log('not mergeable PR, skip');
+    return;
+  }
+
+  const fromSha = pullRequest.base.sha;
+  const toSha = pullRequest.merge_commit_sha;
+  const changedFiles = await getChangedFiles(github, owner, repo, pullNumber);
+
+  const changedFilesContent = await Promise.all(changedFiles.map((f) => async () => {
+    const from = await httpClient.get(rawUrl(owner, repo, fromSha, f.filename));
+    const to = await httpClient.get(rawUrl(owner, repo, toSha, f.filename));
+    return {
+      filepath: f.filename,
+      from: await from.readBody(),
+      to: await to.readBody(),
+    };
+  }).map((f) => f()));
+
+  // by file content
+  const changedLang = maybeTimelineReplaceChange(changedFilesContent);
+  console.log(`changed timelineReplace ${changedLang}`);
+  changedLang.push(...nonNullUnique(lodash.flatten(changedFiles.map((f) => {
+    if (['.js', '.ts'].includes(path.extname(f.filename)))
+      return parseChangedLang(f.patch);
+  }))));
+
+  // by file path
+  const changedModule = nonNullUnique(changedFiles.map((f) => {
+    if (f.filename.startsWith('ui/')) {
+      const [, module] = f.filename.split('/');
+      return module;
+    }
+
+    for (const [prefix, label] of Object.entries(prefixLabelMap)) {
+      if (f.filename.startsWith(prefix))
+        return label;
+    }
+  }));
+
+  return [...changedModule, ...changedLang.map((v) => `💬${v.toUpperCase()}`)];
+};
+
+const getChangedFiles = async (github, owner, repo, pullNumber) => {
+  const listFilesOptions = github.rest.pulls.listFiles.endpoint.merge({
+    owner,
+    repo,
+    // eslint-disable-next-line camelcase
+    pull_number: pullNumber,
+    // this is used by github api, nothing I can do with it.
+  });
+  return await github.paginate(listFilesOptions);
+};
+
+const parseChangedLang = (patch) => {
+  const set = new Set();
+  for (const lang of ['cn', 'de', 'ja', 'fr']) {
+    const pattern = new RegExp(`^\s*[+].*(?:${lang}|'${lang}'): `);
+    for (const line of patch.split('\n')) {
+      if (pattern.test(line))
+        set.add(lang);
+    }
+  }
+  return Array.from(set.values());
+};
+
+const nonNullUnique = (s) => {
+  return lodash.uniq(s).filter(Boolean);
+};
+
+const rawUrl = (owner, repo, sha, path) => {
+  return `https://github.com/${owner}/${repo}/raw/${sha}/${path}`;
+};
+
+const maybeTimelineReplaceChange = (changedFiles) => {
+  const s = new Set();
+
+  changedFiles.forEach((file) => {
+    if (!file.filepath.startsWith('ui/raidboss/data/'))
+      return;
+
+    if (path.extname(file.filepath) === '.js') {
```

Comment:
```suggestion
    if (['.js', '.ts'].includes(path.extname(file.filepath))) {
```

Reply:
I didn't write a parser for ts triggers...

Reply:
And there isn't a TS trigger yet, I can't test it.

Reply:
Ok, we can address this later then.